.github/workflows/build-package.yml

```diff
@@ -21,93 +14,135 @@ on:
         run: |
-          `cygpath "${{ github.workspace }}"`/.github/scripts/download-artifacts.sh ${{ github.run_id }} '${{ inputs.needs }}'
+          `cygpath "${{ github.workspace }}"`/.github/scripts/download-artifacts.sh \
+            ${{ github.run_id }} \
+            '${{ inputs.needs }}'
 
       - name: Install artifacts
-        if: ${{ inputs.needs }}
+        if: inputs.needs
         run: |
           `cygpath "${{ github.workspace }}"`/.github/scripts/install-artifacts.sh
 
       - name: Copy missing headers for mingw-w64-cross-mingwarm64-crt
-        if: ${{ inputs.package_name == 'mingw-w64-cross-mingwarm64-crt' }}
+        if: inputs.package_name == 'mingw-w64-cross-mingwarm64-crt'
         run: |
           `cygpath "${{ github.workspace }}"`/.github/scripts/pthread-headers-hack-before.sh
 
-      - name: Checkout ${{ inputs.packages_repository }} repository
-        uses: actions/checkout@v4
-        with:
-          repository: ${{ inputs.packages_repository }}
-          ref: ${{ inputs.packages_branch }}
-          sparse-checkout: ${{ inputs.package_name }}
-          path: ${{ github.workspace }}/packages
+      - name: Setup MINGWARM64 environment
+        if: inputs.runner_arch == 'aarch64'
+        run: |
+          `cygpath "${{ github.workspace }}"`/.github/scripts/setup-mingwarm64.sh
 
       - name: Enable Ccache
         id: enable-ccache
         run: |
           `cygpath "${{ github.workspace }}"`/.github/scripts/enable-ccache.sh
 
       - name: Restore Ccache
-        uses:  actions/cache/restore@v4
+        uses: actions/cache/restore@v4
         with:
           path: ${{ github.workspace }}/ccache
-          key: main-ccache-${{ steps.enable-ccache.outputs.timestamp }}
-          restore-keys: main-ccache-
+          key: ${{ inputs.package_name }}-ccache-${{ steps.enable-ccache.outputs.timestamp }}
+          restore-keys: ${{ inputs.package_name }}-
 
       - name: Build ${{ inputs.package_name }}
+        timeout-minutes: 720
         working-directory: ${{ github.workspace }}/packages/${{ inputs.package_name }}
         run: |
-          ccache -svv  || true
-          `cygpath "${{ github.workspace }}"`/.github/scripts/build-package.sh ${{ inputs.packages_repository }}
-          ccache -svv || true
+          `cygpath "${{ github.workspace }}"`/.github/scripts/build-package.sh \
+            ${{ inputs.packages_repository }}
 
       - name: Save Ccache
         if: always()
-        uses:  actions/cache/save@v4
+        uses: actions/cache/save@v4
         with:
           path: ${{ github.workspace }}/ccache
-          key: main-ccache-${{ steps.enable-ccache.outputs.timestamp }}
+          key: ${{ inputs.package_name }}-ccache-${{ steps.enable-ccache.outputs.timestamp }}
 
       - name: Upload ${{ inputs.package_name }}
         uses: actions/upload-artifact@v4
         with:
           name: ${{ inputs.package_name }}
-          retention-days: 1
+          retention-days: 3
```

Comment:
The build take around 12 hours and it may fail due to instabilities of the self-hosted runners. For this reason 24 hours retention is not  enough. The size of the artifacts is ~700MB for individual packages and another ~700MB for the whole respository.